src/utils/daos_autotest.c

```diff
@@ -112,6 +118,32 @@ step_init(void)
 	fprintf(autotest_ap->outstream, "Status Time(sec) Comment\033[0m\n");
 }
 
+static int
+attachinfo(void)
+{
+	char	*command = "sudo daos_agent dump-attachinfo";
```

Comment:
sudo? Why is that needed?

Reply:
That's because the command requires read permissions of agent.key file
-bash-4.2$ daos_agent dump-attachinfo
ERROR: daos_agent: Unable to load Certificate Data: could not load key: open /etc/daos/certs/agent.key: permission denied
logger Errorf() failed: write /tmp/daos_agent.log: file already closed

I also believe sudo is used to launch this same command in this other file:
src/tests/ftest/util/agent_utils.py
in this function:
    def dump_attachinfo(self):
        """Run dump-attachinfo on the daos_agent."""
        self.manager.job.set_sub_command("dump-attachinfo")
        self.manager.job.sudo = True

Reply:
This is now removed. End users should not be running daos_agent in this fashion.